src/components/implementation/no_interface/cFE_booter/osfilesys.h

```diff
@@ -1,113 +1,258 @@
+};
+
+// Currently this filedes is used only for dir streams, and not real filedes
+// TODO: Switch non-dir to table based filedes
+struct fd {
+    int32 ino;
+    permission_t access;  /* must be < or == permissive as file->permission */
+    union fd_position position; /* the type of position is determined by file->type */
+    struct fsobj *file;
+};
+
+struct fs {
     char *devname;
     char *volname;
     char *mountpoint;
     uint32 blocksize;
     uint32 numblocks;
     struct fsobj *root;
-    struct FS *next, *prev;
 };
 
+static char *path_to_name(char *path)
+{
+    assert(path);
+    uint32 path_len = strlen(path), offset;
+    assert(path_len > 1);
+
+    //remove one or more '/' at the end of path
+    while(path[strlen(path) - 1] == '/'){
+        path[strlen(path) - 1] = 0;
+    }
+
+    // iterate from right to left through the path until you find a '/'
+    // everything you have iterated through is the name of the file
+    for ( offset = path_len - 2 ; path[offset] != '/' && offset > 0 ; offset--) {
+        //do nothing
+    }
+    char *name = path + offset +1;
+
+    assert(0 < strlen(name));
+    return name;
+}
+
 /******************************************************************************
-** Tar Level Methods
+** fsobj Level Methods
 ******************************************************************************/
+int32 file_rm(struct fsobj *o);
 
-uint32 tar_load();
+int32 file_close_by_ino(int32 ino);
 
-uint32 tar_parse();
+int32 file_close_by_name(char *path);
 
-int32 tar_read(uint32 offset, char *buf, uint32 req_sz);
+uint32 newpart_get(struct f_part **part);
 
-uint32 tar_cphdr(uint32 tar_offset, struct fsobj *file);
+uint32 newfile_get(struct fsobj **o);
 
-uint32 round_to_blocksize(uint32 offset);
+uint32 file_insert(struct fsobj *o, char *path);
 
-static inline uint32 oct_to_dec(char *oct)
-{
-    int32 i, base;
-    int32 tot;
+int32 file_open(char *path, permission_t permission);
 
-    i = strlen(oct) - 1;
+int32 file_close(int32 filedes);
 
-    for (base = 1, tot = 0 ; i >= 0 ; i--, base *= 8) {
-        char val = oct[i];
+int32 file_read(int32 FD, void *buffer, uint32 nbytes);
 
-        assert(val <= '7' && val >= '0');
-        val = val - '0';
-        tot = tot + (val * base);
-    }
+int32 file_write(int32 FD, void *buffer, uint32 nbytes);
 
-    return tot;
-}
+struct fsobj *file_find(char *path);
 
-/******************************************************************************
-** fsobj Level Methods
-******************************************************************************/
+int32 file_create(char *path, permission_t permission);
 
-uint32 fsobj_init(struct fsobj *o);
+int32 file_remove(char *path);
 
-uint32 newfile_get(struct fsobj **o);
+int32 file_rename(char *old_filename, char *new_filename);
+
+int32 file_cp(char *src, char *dest);
+
+int32 file_mv(char *src, char *dest);
 
-uint32 file_insert(struct fsobj *o, struct FS *fs);
+int32 chk_fd(int32 FD);
```

Comment:
I updated with a comment where it is implemented but I don't think it is that ambiguous or worth changing for this pr